frontend/src/pages/search.tsx

```diff
@@ -1,84 +1,218 @@
+import { useEffect, useState } from "react";
 import Layout from "../components/_shared/Layout";
-import TopBar from "../components/_shared/TopBar";
-import { PackageSearchOptions } from "@portaljs/ckan";
-import getConfig from "next/config";
-import { CKAN } from "@portaljs/ckan";
-import DatasetSearchForm from "@/components/search/DatasetSearchForm";
-import DatasetSearchFilters from "@/components/search/DatasetSearchFilters";
-import { env } from "@env.mjs";
-import { MagnifyingGlassIcon, XMarkIcon } from "@heroicons/react/20/solid";
-import { Button } from "@components/ui/button";
-import { SearchIcon } from "lucide-react";
-import DatasetsFilter from "@components/_shared/DatasetsFilter";
 
 import QuickFilterDropdown from "@components/ui/quick-filter-dropdown";
 
-import example from "@data/example.json";
-import datasets from "@data/datasets.json";
-import sectors from "@data/sectors.json";
 import DatasetSearchItem from "@components/search/DatasetSearchItem";
+import SearchBar from "@components/search/SearchBar";
 import {
   Pagination,
   PaginationContent,
-  PaginationEllipsis,
   PaginationItem,
-  PaginationLink,
-  PaginationNext,
-  PaginationPrevious,
 } from "@components/ui/pagination";
+import { Transition } from "@headlessui/react";
+import { cn } from "@lib/utils";
+import { SearchDatasetType } from "@schema/dataset.schema";
+import { api } from "@utils/api";
+import { ChevronLeft, ChevronRight } from "lucide-react";
+import DateQuickFilterDropdown from "@components/ui/date-quick-filter-dropdown";
+import React from "react";
 
-export async function getStaticProps() {
-  const backend_url = env.NEXT_PUBLIC_CKAN_URL;
-  const ckan = new CKAN(backend_url);
-  const search_result = await ckan.packageSearch({
-    offset: 0,
-    limit: 5,
-    tags: [],
-    groups: [],
-    orgs: [],
-  });
-  const groups = await ckan.getGroupsWithDetails();
-  const tags = await ckan.getAllTags();
-  const orgs = await ckan.getOrgsWithDetails(true);
+export function getServerSideProps({ query }: any) {
   return {
-    props: {
-      fallback: {
-        [unstable_serialize([
-          "package_search",
-          { offset: 0, limit: 5, tags: [], groups: [], orgs: [] },
-        ])]: search_result,
-      },
-      groups,
-      tags,
-      orgs,
-    },
+    props: query,
   };
 }
 
+export type SearchPageOnChange = (
+  data: {
+    value: string[] | boolean | string | number | undefined;
+    key: keyof SearchDatasetType;
+  }[]
+) => void;
+
 export default function DatasetSearch({
-  fallback,
-  groups,
-  tags,
-  orgs,
-}: InferGetServerSidePropsType<typeof getStaticProps>): JSX.Element {
-  const router = useRouter();
-  const { q, sector, mode, service, region } = router.query;
-  const [options, setOptions] = useState<PackageSearchOptions>({
+  query,
+  sector,
+  mode,
+  service,
+  region,
+  fuel,
+  before,
+  after,
+  country,
+}: any) {
+  const [modes, setModes] = useState<Facet[]>([]);
```

Comment:
This is considerably more complex than it needs to be, you could've just put everything in here in a `useForm` and that would allow you to keep all the state in sync between the quick and the advanced filters, im not saying to change it, just saying that if you have 12 useStates in a row something is probably wrong